### PR TITLE
Reworking triggered ability text generation to allow for ability words and flavor words to be added more easily

### LINKED ISSUE #7952: Making ability words/flavor words work with triggered abilities
Right now adding an `AbilityWord` to an ability doesn't work if the ability is triggered. Now that we have flavor words in AFR we should probably fix this. I'll probably start it once the rest of the set is done unless someone else wants to do it first.

### PR DESCRIPTION
**Pull request by theelk801**
I still want to go in and do some more refactoring at some point but I think this is a pretty good solution to the issue.

closes #7952

### CODE REVIEW

**Comment by theelk801** [comment]
also check out this gnarly-ass regex I used to do the bulk of the refactoring: 
```
(public\s*String\s*)getRule(\(\)\s*\{\n\s*return\s*)([\s\S]*?)+\s*super.getRule\(\)\;
```

**Comment by JayDi85** [comment]
`getTriggerPhrase` can be removed from `TriggeredAbilityImpl` so it will be required to override in all trigger classes.

**Comment by theelk801** [comment]
I’m not sure if that’s necessary, the idea is that you only use it if you’re not just directly overriding `getRule`

**Comment by JayDi85** [comment]
ok, I've got it -- it's only helps for default behaviour, not `getRules` replace in triggers.

**Comment by theelk801** [comment]
@JayDi85 I'm gonna merge this soon unless you have any objections

File: Mage.Sets/src/mage/cards/h/HallarTheFirefletcher.java [hunk 1]
```diff
@@ -79,7 +79,7 @@ public boolean checkTrigger(GameEvent event, Game game) {
     }
 
     @Override
-    public String getRule() {
-        return "Whenever you cast a spell, if that spell was kicked," + super.getRule();
+    public String getTriggerPhrase() {
+        return "Whenever you cast a spell, if that spell was kicked," ;
```
Comment (JayDi85) [inline Mage.Sets/src/mage/cards/h/HallarTheFirefletcher.java:83]: miss space at the end

File: Mage.Sets/src/mage/cards/v/VraskaSwarmsEminence.java [hunk 2]
```diff
@@ -105,8 +105,8 @@ public boolean checkTrigger(GameEvent event, Game game) {
     }
 
     @Override
-    public String getRule() {
-        return "Whenever a creature you control with deathtouch deals damage to a player or planeswalker," + super.getRule();
+    public String getTriggerPhrase() {
+        return "Whenever a creature you control with deathtouch deals damage to a player or planeswalker," ;
```
Comment (JayDi85) [inline Mage.Sets/src/mage/cards/v/VraskaSwarmsEminence.java:109]: miss space at the end

**Review by JayDi85 — APPROVED** [review]
All fine

**Pull request merged**